client/src/components/ImageUpload/ImageUpload.tsx

```diff
@@ -0,0 +1,61 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+// <input {...state.getInputProps()} /> type DropzoneInputProps is of the wrong type.
+import Dropzone, { DropzoneState } from 'react-dropzone';
+import Button from '@material-ui/core/Button';
+import useStyles from './useStyles';
+import { SetStateAction, useState } from 'react';
+
+interface Props {
+  handleUpload: (files: File[]) => void;
+}
+
+const ImageUpload = ({ handleUpload }: Props): JSX.Element => {
+  const classes = useStyles();
+  const [files, setFiles] = useState([]);
+
+  const onDrop = (acceptedFiles: { map: (arg0: (file: File) => any) => SetStateAction<never[]> }) => {
+    setFiles(
+      acceptedFiles.map((file) =>
+        Object.assign(file, {
+          preview: URL.createObjectURL(file),
+        }),
+      ),
+    );
```

Comment:
Curious, could we limit how many images are uploaded?

Reply:
yes, an additional prop should fix that.